main.tf

```diff
@@ -10,6 +10,14 @@ resource "azurerm_virtual_network" "vnet" {
   address_space       = var.address_space
   dns_servers         = var.dns_servers
   tags                = var.tags
+  dynamic "ddos_protection_plan" {
+    for_each = var.ddos_protection_plan
+
+    content {
+      enable = ddos_protection_plan.value["enable"]
```

Comment:
As the `var.ddos_protection_plan` is a set of `object`, should we access object's field directly here?:

```hcl
enable = ddos_protection_plan.value.enable
```

Reply:
I modified the way to access the value by changing the type of the variable from set(object) to object.